### PR TITLE
[MAINT] update GHA workflow for tests

### LINKED ISSUE #125: automated test in CI: bring them back online
Can have a quick look into it.

### LINKED ISSUE #126: modernize packaging
in short get away from setup.py and move to pyproject.toml

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Fixes #125 
- Fixes #126

Changes proposed in this pull request:
- add dependabot config to keep github actions updated on a monthly basis
- test package on all OS and most recent python version

### CODE REVIEW

**Comment by Remi-Gau** [comment]
Note: tuto smoke test passes on windows but fails on other OS

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+[build-system]
+build-backend = "hatchling.build"
+requires = ["hatchling"]
+
+[project]
+classifiers = [
+    'Development Status :: 5 - Production/Stable',
```
Comment (Remi-Gau) [inline pyproject.toml:7]: Copied this one but I am not entirely sure this is true.

**Comment by Remi-Gau** [comment]
> Note: tuto smoke test passes on windows but fails on other OS

Actually that's a lie. Tuto fails on windows but is not reported as such in CI.

https://github.com/BrkRaw/brkraw/actions/runs/6192345440/job/16812264179?pr=134#step:6:21


```
============================= test session starts =============================
platform win32 -- Python 3.8.10, pytest-7.4.2, pluggy-1.3.0
rootdir: D:\a\brkraw\brkraw
plugins: nbmake-1.4.3
collected 1 item

tests\tutorials\JupyterNotebooks\01_GettingStarted.ipynb F               [100%]

================================== FAILURES ===================================
_ D:\a\brkraw\brkraw\tests\tutorials\JupyterNotebooks\01_GettingStarted.ipynb _
---------------------------------------------------------------------------
# the default metadata reference.
pp.pprint(brkraw.lib.reference.METADATA_FILED_INFO)
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[8], line 2
      1 # the default metadata reference.
----> 2 pp.pprint(brkraw.lib.reference.METADATA_FILED_INFO)

AttributeError: module 'brkraw.lib.reference' has no attribute 'METADATA_FILED_INFO'
=========================== short test summary info ===========================
FAILED tests/tutorials/JupyterNotebooks/01_GettingStarted.ipynb:: - AttributeError: module 'brkraw.lib.reference' has no attribute 'METADATA_FILED_INFO'
============================== 1 failed in 4.22s ==============================
Folder PATH listing for volume Temporary Storage
Volume serial number is 60C0-CA7D
D:\A\BRKRAW\BRKRAW\TESTS\TUTORIALS\RAW
Invalid path - \A\BRKRAW\BRKRAW\TESTS\TUTORIALS\RAW
No subfolders exist 
```

**Comment by gdevenyi** [comment]
I'm not super inclined to worry about windows right now. I'd rather improve the overall Unix process and come back later, thoughts @Remi-Gau?

**Comment by Remi-Gau** [comment]
You decide.
Main concern would be if most users are based on windows then maybe worth it.
But maybe not at first.

**Pull request merged**